File: engine/model/src/main/scala/tools/packet/playerinteraction/PlayerShopRemoveVisitor.scala
package tools.packet.playerinteraction

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PlayerShopRemoveVisitor(private var _slot: Int) extends PacketInput {
  def slot: Int = _slot

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/PlayerShopRemoveVisitor.java
package tools.packet.character.interaction;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PlayerShopRemoveVisitor(Integer slot) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}